doc/source/serve/advanced-guides/grpc-guide.md

```diff
@@ -0,0 +1,247 @@
+(serve-set-up-grpc-service)=
+# Set Up gRPC Service
+
+This section helps you understand how to:
+- build a user defined gRPC service and protobuf
+- start Serve with gRPC proxy
+- deploy gRPC applications
+- send gRPC requests to Serve deployments
+- health checks proxies
+- gRPC request metadata 
+- more examples
+- error handling
+
+
+(custom-serve-grpc-service)=
+## Define a gRPC Service
+To run a gRPC server it starts with first defining gRPC services, rpc methods, and 
+protobufs similar to the one below.
+
+```{literalinclude} ../doc_code/proxy_guide/user_defined_protos.proto
+:start-after: __begin_proto__
+:end-before: __end_proto__
+:language: proto
+```
+
+
+In this example, we created a file named `user_defined_protos.proto`. There are two
+gRPC services, `UserDefinedService` and `FruitService`. `UserDefinedService` has three
+RPC methods, `__call__`, `Multiplexing`, and `Streaming`. `FruitService` has one
+RPC method, `FruitStand`. Their corresponding input and output types are also defined
+specifically for each RPC method.
```

Comment:
Can we use a more realistic example instead of fruit stand? Maybe image detection or text generation?

We've had feedback to remove this from the docs internally and externally. So let's not add new ones using it.